WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java

```diff
@@ -183,17 +230,26 @@ public Object execute(RequestForm requestForm, BindException errors) throws Exce
                 throw new BatchValidationException(validationExceptions, null);
             }
 
+            //email notification
             EmailTemplateForm emailTemplateForm = getValuesForEmailTemplate(requestForm);
 
-            //if its a new request
-            if (null != requestForm.getIsNewRequest() && requestForm.getIsNewRequest())
+            try
             {
-                sendNewRequestEmailNotification(emailTemplateForm);
+                //if its a new request
+                if (null != requestForm.getIsNewRequest() && requestForm.getIsNewRequest())
+                {
+                    sendNewRequestEmailNotification(emailTemplateForm);
+                }
+                else
+                {
+                    sendRequestChangeEmailNotification(oldStatusVal, emailTemplateForm, requestForm, oldLineItems);
+                }
             }
-            else
+            catch (RuntimeException | MessagingException | IOException | ValidationException e)
```

Comment:
This seems overly broad. A failure to send emails shouldn't cause ValidationException or NPEs, should it?

Reply:
Yes, however, NotificationService.get().sendMessageForRecipient throws all three exceptions

Reply:
Looking at the sendMessageForRecipient, I don't really like how it's using ValidationException but catching it here seems OK. I'm more worried about the RuntimeException part of the catch - were you getting any of those thrown in your development?

Reply:
I think this note got missed. I recommend removing RuntimeException from this list. And also logging the full exception instead of just the message in the _log.error() call below.

Reply:
Done!